paima-demo/README.md

```diff
@@ -0,0 +1,88 @@
+# Godot-cardano and Paima's "Open World" prototype
+
+- [Godot-cardano and Paima's "Open World" prototype](#godot-cardano-and-paimas-open-world-prototype)
+  - [Prerequisites](#prerequisites)
+  - [Setting up the demo](#setting-up-the-demo)
+    - [Add `addons`](#add-addons)
+    - [Compile WASM](#compile-wasm)
+    - [Generate and setup Paima template and Batcher](#generate-and-setup-paima-template-and-batcher)
+    - [Export Godot demo project](#export-godot-demo-project)
+  - [Running the demo](#running-the-demo)
+    - [Start Paima node and required infrastructure](#start-paima-node-and-required-infrastructure)
+    - [Starting frontend](#starting-frontend)
+  - [Notes](#notes)
+  - [TODO](#todo)
+
+This is combination of Paima's "open-world" template and Godot project that serves for testing interactions between web-exported Godot project and Paima middleware.
+
+## Prerequisites
+
+- `Docker` (required to start testnet an other infra for Paima)
+- `paima-engine`. Tested with `paima-engine-linux-2.3.0`).
+- Rust (nightly). Tested with `cargo 1.77.1 (e52e36006 2024-03-26)`.
+- `emscripten` for WASM and WEB-export. See [Godot-rust book](https://godot-rust.github.io/book/toolchain/export-web.html). Tested with `emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.39 (36f871819b566281d160470a1ec4515f379e9f77)`)
+- `node` and `npm` - used a lot by Paima setup (see [Makefile](./Makefile) and generated Paima READMEs). Tested with `node v20.12.2` and `npm 10.5.0`
+- Godot. Tested with `Godot_v4.2.1-stable_linux.x86_64` (extra flag that I had tu use: `--rendering-driver opengl3`). [WEB-export docs](https://docs.godotengine.org/en/stable/tutorials/export/exporting_for_web.html)
+- `python` - to run web-server with required CORS headers (script taken from [godot manual `Tip` section](https://docs.godotengine.org/en/stable/tutorials/export/exporting_for_web.html#serving-the-files))
+
+`node`, `npm` and `python` are available via `flake.nix` in the root of the current dir.
```

Comment:
I'm not very good at Nix, so kept my simple `flake.lock` here instead of messing the project's one. Node and npm are used by Paima setup extensively for everything: init projects, generate files and even to start Docker containers for testnet. python is only required to start web-server to serve frontend - can be substituted by anything else; script has examples of headers that need to be set on webserver side.